memory-db/src/lib.rs

```diff
@@ -107,36 +113,48 @@ impl<T> MaybeDebug for T {}
 ///   assert!(!m.contains(&k, EMPTY_PREFIX));
 /// }
 /// ```
-pub struct MemoryDB<H, KF, T>
-	where
+pub struct MemoryDB<H, KF, T, M = CountingCallback<T>>
+where
 	H: KeyHasher,
 	KF: KeyFunction<H>,
+	M: MallocSizeOfCallback<T>,
```

Comment:
To me the word "callback" is a single callable function/closure, not a collection of callbacks. Maybe this should be named "MallocSizeTracker" or "MemTracker"? Or "MallocSizeOfCallbacks" (plural)? Using "callback" here puts the focus on how the tracking is done, rather than what this entity really is: a means to track memory usage.